nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -6,27 +6,27 @@ let
   cfg = config.services.nextcloud;
   fpm = config.services.phpfpm.pools.nextcloud;
 
-  phpPackage = pkgs.php73;
-  phpPackages = pkgs.php73Packages;
+  phpPackage = pkgs.php74.buildEnv {
+    exts = pp: with pp.exts; [
+      bcmath calendar curl exif ftp filter gd gettext gmp intl json ldap
+      mysqlnd opcache openssl pcntl pdo pdo_mysql pdo_odbc pdo_pgsql
+      pdo_sqlite pgsql readline session soap sodium sqlite3 zip zlib mbstring
+      posix ctype dom simplexml xmlreader xmlwriter pp.apcu
+      pp.redis pp.memcached pp.imagick
+    ];
+    extraConfig = phpOptionsStr;
+  };
 
   toKeyValue = generators.toKeyValue {
     mkKeyValue = generators.mkKeyValueDefault {} " = ";
   };
 
-  phpOptionsExtensions = ''
-    ${optionalString cfg.caching.apcu "extension=${phpPackages.apcu}/lib/php/extensions/apcu.so"}
-    ${optionalString cfg.caching.redis "extension=${phpPackages.redis}/lib/php/extensions/redis.so"}
-    ${optionalString cfg.caching.memcached "extension=${phpPackages.memcached}/lib/php/extensions/memcached.so"}
-    extension=${phpPackages.imagick}/lib/php/extensions/imagick.so
-    zend_extension = opcache.so
-    opcache.enable = 1
```

Comment:
Is `opcache.enable = 1` not needed anymore? This isn't part of `extraConfig` anymore AFAICS.

Reply:
Opcache is enabled by default if the module is loaded, according to the [PHP website](https://www.php.net/manual/en/opcache.configuration.php). I'll check if this is the case when running the binary.

Reply:
It is indeed on by default.